main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+func getInput(prompt string, r *bufio.Reader) (string, error) {
+	fmt.Print(prompt)
+	input, err := r.ReadString('\n')
+
+	return strings.TrimSpace(input), err
+}
+
+func generatePlaylist() (track, artist, []relatedartist) {
```

Comment:
This is fine for now as the project is small, but the `generatePlaylist` function is doing too many things. Right now it's handling the user's input, the actual algorithm for creating a playlist based from that input, and the display to the user.

Probably would be best to separate out the stages, so first have some function to parse the user's input, then change the generatePlaylist function to take a song's id as its parameter, generate the playlist and return. Then finally have some other function that displays (or saves, shares, whatever) the playlist to the user.

Reply:
agreed. at the moment i've just been using the `generatePlaylist` as a scratch pad which has changed with every iteration, first just dumping out the track info from the id, then various other data and now doing this. so the plan was to separate this out once the algorithm gets more complex so that it's not doing multiple things. i'll separate it out now though, for correctness